src/main/java/com/google/cloud/tools/app/InvalidDirectoryException.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.app;
+
+/**
+ * Signals an invalid directory.
+ */
+public class InvalidDirectoryException extends RuntimeException {
```

Comment:
You can keep them as their own exceptions if you want to catch them
specifically, just subclass IAE instead of RuntimeException.

On Thu, Mar 24, 2016 at 5:48 PM, João Martins <notifications@github.com>
wrote:

> In src/main/java/com/google/cloud/tools/app/InvalidDirectoryException.java
> <https://github.com/GoogleCloudPlatform/app-tools-lib-for-java/pull/1#discussion_r57393468>
> :
>
> > + * You may obtain a copy of the License at
> > + *
> > + *     http://www.apache.org/licenses/LICENSE-2.0
> > + *
> > + * Unless required by applicable law or agreed to in writing, software
> > + * distributed under the License is distributed on an "AS IS" BASIS,
> > + * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + */
> > +package com.google.cloud.tools.app;
> > +
> > +/**
> > + * Signals an invalid directory.
> > + */
> > +public class InvalidDirectoryException extends RuntimeException {
>
> At the beginning, I had a special treatment for these in the clients, but
> that was a long way back, so I'll get rid of them and use IAEs.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/app-tools-lib-for-java/pull/1/files/a4df7ff1686048e2ffd9a30d1f8ceb37805a64d6#r57393468>
>